crates/ruff_linter/src/rules/refurb/rules/fstring_number_format.rs

```diff
@@ -134,15 +132,57 @@ pub(crate) fn fstring_number_format(checker: &Checker, subscript: &ast::ExprSubs
     );
 
     if let Some(replacement) = replacement {
-        diagnostic.set_fix(Fix::safe_edit(Edit::range_replacement(
-            replacement,
-            subscript.range(),
-        )));
+        let edit = Edit::range_replacement(replacement, subscript.range());
+        diagnostic.set_fix(Fix::applicable_edit(edit, applicability));
     }
 
     checker.report_diagnostic(diagnostic);
 }
 
+/// Generate a replacement, if possible.
+fn try_create_replacement(checker: &Checker, arg: &Expr, base: Base) -> Option<String> {
+    if !matches!(
+        arg,
+        Expr::NumberLiteral(_) | Expr::Name(_) | Expr::Attribute(_)
+    ) {
+        return None;
+    }
+
+    let inner_source = checker.locator().slice(arg);
+
+    // On Python 3.11 and earlier, trying to replace an `arg` that contains a backslash
+    // would create a `SyntaxError` in the f-string.
+    if checker.target_version() <= PythonVersion::PY311 && inner_source.contains('\\') {
+        return None;
+    }
+
+    // On Python 3.11 and earlier, trying to replace an `arg` that spans multiple lines
+    // would create a `SyntaxError` in the f-string.
+    if checker.target_version() <= PythonVersion::PY311 && inner_source.contains('\n') {
```

Comment:
To check for multiple lines, `find_newline` is better because it also detects `'\r'`.